core/src/main/java/org/apache/spark/shuffle/unsafe/UnsafeShuffleExternalSorter.java

```diff
@@ -86,6 +86,9 @@
 
   private final LinkedList<SpillInfo> spills = new LinkedList<SpillInfo>();
 
+  // Peak memory used by this sorter, set immediately before each spill
```

Comment:
What if we never spill?  In that case, will we report the peak memory usage as zero?